### PR TITLE
Feature/support proxy and user agent

### PR DESCRIPTION
**Pull request by tetsuco**

### CODE REVIEW

File: mercapi/mercapi.py [hunk 1]
```diff
@@ -22,19 +24,28 @@ class Mercapi:
...
-        "User-Agent": _user_agent,
-        "X-Platform": "web",
-    }
-
-    def __init__(self):
+    
+    def __init__(
+        self,
+        *,
+        proxies: typing.Optional[ProxiesTypes] = None,
```
Comment (take-kun) [inline mercapi/mercapi.py]: `Optional` is imported directly with `from typing import Optional`. I don't understand why you'd re-import the entire `typing` module to use `typing.Optional` here.

File: mercapi/mercapi.py [hunk 2]
```diff
@@ -22,19 +24,28 @@ class Mercapi:
...
+        user_agent: str = (
```
Comment (take-kun) [inline mercapi/mercapi.py]: Also, please follow the convention of optional parameters, and set `None` as a default value here. Then, the constructor should set the default UA if `user_agent` parameter is `None`.

```python
if not user_agent:
    self._headers = { "User-Agent": "Mozilla/5.0 ..." }
```

File: mercapi/mercapi.py [hunk 3]
```diff
@@ -22,19 +24,28 @@ class Mercapi:
...
+            "Mozilla/5.0 (X11; Linux x86_64; rv:102.0) Gecko/20100101 Firefox/102.0"
+        ),
+    ):
+        """initialize
+
+        :param proxies: Proxy. ex) {"http://": "http://localhost:8080", "https://": "http://localhost:8080"}
```
Comment (take-kun) [inline mercapi/mercapi.py]: Thanks for taking care of documenting your changes, although I wouldn't mind if this were a bit more descriptive. Please let me know if you need help with it.

**Review by take-kun — CHANGES_REQUESTED** [review]

File: mercapi/mercapi.py (hunk 1 above)
Comment (tetsuco) [reply mercapi/mercapi.py]: Sorry, mistake. Corrected.

File: mercapi/mercapi.py (hunk 2 above)
Comment (tetsuco) [reply mercapi/mercapi.py]: Thanks for pointing that out. I have corrected it that way, is that ok?

File: mercapi/mercapi.py (hunk 3 above)
Comment (tetsuco) [reply mercapi/mercapi.py]: I wrote a little more carefully.

**Review by take-kun — APPROVED** [review]

**Comment by take-kun** [comment]
sorry it took me a while, i was busy with other things

**Pull request merged**